src/System.ComponentModel.Annotations/tests/System/ComponentModel/DataAnnotations/AssociationAttributeTests.cs

```diff
@@ -14,33 +14,32 @@ public class AssociationAttributeTests
         [InlineData(null, "", " \t \r \n", new string[] { "" }, new string[] { "\t\r\n" })]
         [InlineData(null, null, null, new string[0], new string[0])]
         [InlineData("Name", "ThisKey1,  ThisKey2, ThisKey3", "OtherKey1,  OtherKey2",new string[] { "ThisKey1", "ThisKey2", "ThisKey3" }, new string[] { "OtherKey1", "OtherKey2" })]
-        public static void Constructor(string name, string thisKey, string otherKey, string[] thisKeyMembers, string[] otherKeyMembers)
+        public void Ctor_String_String_String(string name, string thisKey, string otherKey, string[] expectedThisKeyMembers, string[] expectedOtherKeyMembers)
         {
             var attribute = new AssociationAttribute(name, thisKey, otherKey);
             Assert.Equal(name, attribute.Name);
             Assert.Equal(thisKey, attribute.ThisKey);
             Assert.Equal(otherKey, attribute.OtherKey);
-
-            if (thisKey == null)
+            if (PlatformDetection.IsFullFramework && thisKey == null)
             {
                 Assert.Throws<NullReferenceException>(() => attribute.ThisKeyMembers);
             }
             else
             {
-                Assert.Equal(thisKeyMembers, attribute.ThisKeyMembers);
+                Assert.Equal(expectedThisKeyMembers, attribute.ThisKeyMembers);
             }
-            if (otherKey == null)
+            if (PlatformDetection.IsFullFramework && otherKey == null)
```

Comment:
Aside. We probably should delete our tests and test behaviors that are specific to .NET Framework. It never was and is never going to be a goal to use CoreFX tests to find bugs in .NET Framework. Really their only value is as documentation, where we intentionally changed behavior, there are two similar tests next to each other. This is an expensive kind of documentation, that runs every day, may sporadically fail and need analysis, etc. Comments would be better. Either way is fine here for now but in general when we delete them it will be easier if they are whole tests and not branches within tests.

That isn't the case for running tests that apply the same to both. There is still value in new tests defaulting to running on both. But more and more often, new tests will not apply to .NET Framework because new tests will mostly be for new API and behaviors. At some point, later still, we will likely decide the existing tests are enough to preserve compat and the .NET Framework runs are not worth the cost. Or at least, change the default so that new tests default to .NET Core only.

Reply:
@karelz @geoffkizer @stephentoub 

>At some point, later still, we will likely decide the existing tests are enough to preserve compat and the .NET Framework runs are not worth the cost. Or at least, change the default so that new tests default to .NET Core only.

This is an interesting discussion.  We have discussed this within networking recently.  For example, when writing tests to support HTTP/2 behavior in SocketsHttpHandler/HttpClientHandler, do we skip running the tests on .NET Framework.  Or even skip running them on the other HTTP stacks (WinHttpHandler, CurlHandler).  We haven't yet converged on a final consensus.  We do have a lot of great test collateral now in CoreFx and it's useful for finding unexpected behavior differences between .NET Framework and .NET Core.

Reply:
> We do have a lot of great test collateral now in CoreFx and it's useful for finding unexpected behavior differences between .NET Framework and .NET Core.

I agree. The conditional logic here can be painful and messy, but it's been surprisingly valuable.

Reply:
Interesting. I opened https://github.com/dotnet/corefx/issues/35088 to discuss this. I certainly don't want to eliminate anything with a positive cost/benefit. Feel free to chime in there. I'd like to understand a bit better what the value is. Eg., you are still adding new tests for behavior common to .NET Framework as well - I can see the value in that case.

Reply:
Another possible reason - you find it useful for verifying servicing changes in .NET Framework, because the .NET Framework's own tests are lacking (although in that case a frozen set of tests might be as good)